contracts/colony/ColonyDataTypes.sol

```diff
@@ -147,57 +147,81 @@ interface ColonyDataTypes {
   /// @param amount Amount of the payout funding
   event TaskPayoutSet(uint256 indexed taskId, TaskRole role, address token, uint256 amount);
 
+  /// @notice Event logged when task data is changed via signed messages by those involved
+  /// @param reviewerAddresses Array of addresses that signed off this change.
+  event TaskChangedViaSignatures(address[] reviewerAddresses);
+
   /// @notice Event logged when a deliverable has been submitted for a task
   /// @param taskId Id of the task
   /// @param deliverableHash Hash of the work performed
-  event TaskDeliverableSubmitted(uint256 indexed taskId, bytes32 deliverableHash);
+  event TaskDeliverableSubmitted(address agent, uint256 indexed taskId, bytes32 deliverableHash);
 
   /// @notice Event logged when a task has been completed. This is either because the dueDate has passed
   /// and the manager closed the task, or the worker has submitted the deliverable. In the
   /// latter case, TaskDeliverableSubmitted will also be emitted.
-  event TaskCompleted(uint256 indexed taskId);
+  event TaskCompleted(address agent, uint256 indexed taskId);
 
   /// @notice Event logged when the rating of a role was revealed
   /// @param taskId Id of the task
   /// @param role Role that got rated
   /// @param rating Rating the role received
-  event TaskWorkRatingRevealed(uint256 indexed taskId, TaskRole role, uint8 rating);
+  event TaskWorkRatingRevealed(address agent, uint256 indexed taskId, TaskRole role, uint8 rating);
 
   /// @notice Event logged when a task has been finalized
   /// @param taskId Id of the finalized task
-  event TaskFinalized(uint256 indexed taskId);
+  event TaskFinalized(address agent, uint256 indexed taskId);
 
   /// @notice Event logged when tokens are minted
   /// @param who The address being awarded the tokens
   /// @param amount The amount of tokens being awarded
-  event TokensMinted(address who, uint256 amount);
+  event TokensMinted(address agent, address who, uint256 amount);
 
   /// @notice Event logged when a payout is claimed, either from a Task or Payment
   /// @param fundingPotId Id of the funding pot where payout comes from
   /// @param token Token of the payout claim
   /// @param amount Amount of the payout claimed, after network fee was deducted
-  event PayoutClaimed(uint256 indexed fundingPotId, address token, uint256 amount);
+  event PayoutClaimed(address agent, uint256 indexed fundingPotId, address token, uint256 amount);
 
   /// @notice Event logged when a task has been canceled
   /// @param taskId Id of the canceled task
   event TaskCanceled(uint256 indexed taskId);
```

Comment:
Why no change here?

Reply:
Cancelling a task goes through the signed-by-contributors process that results in `msg.sender` in that call always being the colony. The transaction now [emits](https://github.com/JoinColony/colonyNetwork/pull/911/files#diff-22a3ddcee839f8d7c9a853e0226611bf413c43506cc778a24d701c420f3b1871R1447) the new `TaskChanvedViaSignatures` event as well with the relevant addresses.